packages-exp/auth-compat-exp/src/user.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as impl from '@firebase/auth-exp';
+import { UserImpl } from '@firebase/auth-exp/src/core/user/user_impl';
+import { UserParameters } from '@firebase/auth-exp/src/model/user';
+import * as compat from '@firebase/auth-types';
+import * as externs from '@firebase/auth-types-exp';
+import '@firebase/installations';
+import {
+  convertComfirmationResult,
+  convertCredential
+} from './user_credential';
+
+export class User extends UserImpl implements compat.User {
+  readonly multiFactor: compat.MultiFactorUser;
+
+  constructor(params: UserParameters) {
+    super(params);
+    this.multiFactor = impl.multiFactor(this);
+  }
+
+  getIdTokenResult(forceRefresh?: boolean): Promise<compat.IdTokenResult> {
+    return impl.getIdTokenResult(this, forceRefresh);
+  }
+  getIdToken(forceRefresh?: boolean): Promise<string> {
+    return impl.getIdToken(this, forceRefresh);
+  }
+  linkAndRetrieveDataWithCredential(
+    credential: compat.AuthCredential
+  ): Promise<compat.UserCredential> {
+    return this.linkWithCredential(credential);
```

Comment:
why is this a method?

Reply:
Defined (albeit deprecated) in the old SDK https://github.com/firebase/firebase-js-sdk/blob/master/packages/auth-types/index.d.ts#L27